astropy/coordinates/baseframe.py

```diff
@@ -156,6 +156,67 @@ def __new__(mcls, name, bases, members):
                 'Could not find all expected BaseCoordinateFrame class '
                 'attributes.  Are you mis-using FrameMeta?')
 
+        # Unless overridden, velocity name defaults are:
+        #   * pm_{lon}_cos{lat} for SphericalCosLatDifferential
+        #   * radial_velocity for any d_distance, including RadialDifferential
+        #   * v_{x,y,z} for d_{x,y,z} for CartesianDifferential
+        # where {lon} and {lat} are the names of the angular components
+        if repr_info is None:
+            repr_info = {}
```

Comment:
If `repr_info` is not `None`, I think you need to copy it here, otherwise you're mucking with the contents.

Though... since this metaclass replaces what was put in anyway, it would seem to make the problem you mentioned even worse.

But your solution surely is on the right track: we want to define this stuff right at class-creation time, so that the instance's `_get_representation_info()` can be very fast.

Actually... Might that be another solution? Here, pre-calculate a *new* dict that is used by `_get_representation_info()`?

Must admit that everything seems overly tricky; unfortunately, no real time to think until the end of the week.

Reply:
The copy now happens up on L150.